packages/drivekit_trip_analysis/flutter_drivekit_trip_analysis_android/lib/src/adapter.dart

```diff
@@ -27,3 +27,466 @@ extension PigeonAdapter on Vehicle {
+      gpsElevation: gpsElevation.toList(),
+      gpsAccuracy: gpsAccuracy.toList(),
+      gpsHeading: gpsHeading.toList(),
+      screenLocked: screenLocked.toList(),
+      activityValue: activityValue.toList(),
+      roll: roll.toList(),
+      pitch: pitch.toList(),
+      yaw: yaw.toList(),
+      gyroscopeNormVar: gyroscopeNormVar.toList(),
+    );
+  }
+}
+
+/// Adapts the [PostGenericResponse] class to the corresponding Pigeon
+extension PigeonPostGenericResponseAdapter on PostGenericResponse {
+  /// Converts a [PigeonPostGenericResponse] to a corresponding model class.
+  PigeonPostGenericResponse toPigeonImplementation() {
+    return PigeonPostGenericResponse(
+      status: status,
+      itinId: itinId,
+      comments: comments.map((e) => e.toPigeonImplementation()).toList(),
+      userId: userId,
+      itineraryStatistics: itineraryStatistics?.toPigeonImplementation(),
+      ecoDriving: ecoDriving?.toPigeonImplementation(),
+      fuelEstimation: fuelEstimation?.toPigeonImplementation(),
+      safety: safety?.toPigeonImplementation(),
+      advancedEcoDriving: advancedEcoDriving?.toPigeonImplementation(),
+      advancedFuelEstimation: advancedFuelEstimation?.toPigeonImplementation(),
+      advancedSafety: advancedSafety?.toPigeonImplementation(),
+      pollutants: pollutants?.toPigeonImplementation(),
+      tireWear: tireWear?.toPigeonImplementation(),
+      brakeWear: brakeWear?.toPigeonImplementation(),
+      driverDistraction: driverDistraction?.toPigeonImplementation(),
+      itineraryData: itineraryData?.toPigeonImplementation(),
+      endDate: endDate,
+      logbook: logbook?.toPigeonImplementation(),
+      safetyEvents:
+          safetyEvents?.map((e) => e.toPigeonImplementation()).toList(),
+      callEvents: callEvents?.map((e) => e.toPigeonImplementation()).toList(),
+      speedingEvents:
+          speedingEvents?.map((e) => e.toPigeonImplementation()).toList(),
+      speedingStatistics: speedingStatistics?.toPigeonImplementation(),
+      energyEstimation: energyEstimation?.toPigeonImplementation(),
+      advancedEnergyEstimation: advancedEnergyEstimation
+          ?.map((e) => e.toPigeonImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [DriverDistraction] class to the corresponding Pigeon
+extension PigeonDriverDistractionAdapter on DriverDistraction {
+  /// Converts a [DriverDistraction] to a corresponding Pigeon.
+  PigeonDriverDistraction toPigeonImplementation() {
+    return PigeonDriverDistraction(
+      nbUnlock: nbUnlock,
+      durationUnlock: durationUnlock,
+      durationPercentUnlock: durationPercentUnlock,
+      distanceUnlock: distanceUnlock,
+      distancePercentUnlock: distancePercentUnlock,
+      score: score,
+      scoreUnlock: scoreUnlock,
+      scoreCall: scoreCall,
+      calls: calls?.map((e) => e.toPigeonImplementation()).toList(),
+    );
+  }
+}
+
+/// Adapts the [PigeonCall] class to the corresponding class in the model.
```

Comment:
"/// Adapts the [**Call**] class to the corresponding **Pigeon**"

Reply:
corrigé sur 7cecc2affa21960d1b1c86074dea479cc4822f35